lib/Conversion/TorchToTosa/TosaLegalizeCommon.cpp

```diff
@@ -268,7 +268,8 @@ convertReduceMeanOp(PatternRewriter &rewriter, Operation *op,
     int64_t axis_val = axes_elems.getValues<IntegerAttr>()[i].getInt();
     if (axis_val < 0)
       axis_val += input_rank;
-    num_elems_on_reduced_axis *= input_type.getShape()[axis_val];
+    int64_t inShape = input_type.getShape()[axis_val];
```

Comment:
@eric-k256 it seems nonsensical to multiply by -1 here. Can we fix this?

Reply:
The short term fix to handle this is to fail legalization if reducing over a dynamic axis. The rest of the legalization is assuming that you know how many elements you're calculating the mean of for the MUL at the end of the reduction.

Reply:
I don't understand the rationale behind this, but this kind of product of the dims (even with the negative values) has been done at many other places as well in the `TorchToTosa`.